openedx/core/lib/api/authentication.py

```diff
@@ -5,11 +5,10 @@
 import django.utils.timezone
 from oauth2_provider import models as dot_models
 from provider.oauth2 import models as dop_models
-from rest_framework import exceptions as drf_exceptions
+from rest_framework.exceptions import AuthenticationFailed
```

Comment:
The new version of DRF allows details to be passed as dicts, which was the whole reason for the custom layer of exceptions wrapping the DRF exceptions. So I removed the entire layer and shifted the code to use/handle the straight DRF exceptions. This change is larger than I wanted to make - but it seemed like the right thing to do.